src/domain/core/Entity.ts

```diff
@@ -1,12 +1,19 @@
-import type { ValueObject } from '@/domain/core/ValueObject'
-import { isNull, isUndefined } from 'lodash'
+import type {ValueObject} from '@/domain/core/ValueObject'
+import {isNull, isUndefined} from 'lodash'
+
+export class EntityCorrupted extends Error {
```

Comment:
I quite not understand this EntityCorrupted 
what is the difference between it and ValueFailure 
when should we use it?

Reply:
NOTE: after answer the question, I think I can improve it without EntityCorrupted. By making `low` and ValueFailure in case it is missing in data. Read on.

I was thinking this way Entity contain multiple value let's say:
```
BloodPressure {
  high: BeatPerMinuteValue,
  low: BeatPerMinuteValue,
}
```

Let's example some data flow in:
Case 1:
```
{
  high: 20,
  // low: is missing
}
```
=> Entity corrupted

Case 2:
```
{
  high: 20,
  low: -10, // Defined in BeatPerMinuteValue that it must be a number > 30 and < 250
```
=> Entity can be parsed, but value `low` is a ValueFailure

Reply:
I got it 
thank you for your explanation :)